tensorflow/python/lib/io/file_io_test.py

```diff
@@ -62,6 +62,52 @@ def testEmptyFilename(self):
     with self.assertRaises(errors.NotFoundError):
       _ = f.read()
 
+  def testJoinUrlLike(self):
+    """file_io.join joins url-like filesystems with '/' on all platform.
+    """
+    for fs in ("ram://", "gcs://"):
+      expected = fs + 'exists/a/b/c.txt'
+      self.assertEqual(
+        file_io.join(fs, 'exists', 'a', 'b', 'c.txt'),
+        expected
+      )
+      self.assertEqual(
+        file_io.join(fs + 'exists', 'a', 'b', 'c.txt'),
+        expected
+      )
+      self.assertEqual(
+        file_io.join(fs, 'exists/a', 'b', 'c.txt'),
+        expected
+      )
+      self.assertEqual(
+        file_io.join(fs, 'exists', 'a', 'b/c.txt'),
+        expected
+      )
+
+  def testJoinFilesystem(self):
+    """file_io.join respects the os.path.join behavior for native filesystems.
+    """
```

Comment:
@mihaimaruseac let me know if this is the desired behavior. I realize that `file://` is the native filesystem, but maybe it should be treated like other ulr-like filesystems? What does `GFile(join("file://", "dir", "file.py"))` expect?

Reply:
I think it should be `"file://dir/file.py"`. It's similar to URLs

Reply:
easy enough change; and it simplifies the implementation/testing.

Reply:
@mihaimaruseac I just pushed this change, please re-approve so CI runs when you get a chance 😃